src/lib/Behat/Component/UserNotificationPopup.php

```diff
@@ -41,6 +47,140 @@ public function clickNotification(string $expectedType, string $expectedDescript
         throw new Exception(sprintf('Notification of type: %s with description: %d not found', $expectedType, $expectedDescription));
     }
 
+    public function verifyNotification(string $expectedType, string $expectedAuthor, string $expectedDescription, ?string $expectedDate = null, bool $shouldExist = true): void
+    {
+        $notifications = $this->getHTMLPage()->setTimeout(5)->findAll($this->getLocator('notificationItem'));
+
+        foreach ($notifications as $notification) {
+            $criteria = [
+                new ChildElementTextCriterion($this->getLocator('notificationType'), $expectedType),
+                new ChildElementTextCriterion($this->getLocator('notificationDescriptionTitle'), $expectedAuthor),
+                new ChildElementTextCriterion($this->getLocator('notificationDescriptionText'), $expectedDescription),
+            ];
+
+            if ($expectedDate !== null && $expectedDate !== 'XXXX-XX-XX') {
+                $criteria[] = new ChildElementTextCriterion($this->getLocator('notificationDate'), $expectedDate);
+            }
+
+            foreach ($criteria as $criterion) {
+                if (!$criterion->matches($notification)) {
+                    continue 2; // go to next notification
+                }
+            }
+
+            if ($shouldExist) {
+                return; // matching notification found
+            } else {
+                throw new \Exception(sprintf(
+                    'Notification of type "%s" with author "%s" and description "%s" should not exist, but was found.',
+                    $expectedType,
+                    $expectedAuthor,
+                    $expectedDescription
+                ));
+            }
+        }
+
+        if ($shouldExist) {
+            throw new \Exception(sprintf(
+                'Notification of type "%s" with author "%s" and description "%s" was not found.',
+                $expectedType,
+                $expectedAuthor,
+                $expectedDescription
+            ));
+        }
+    }
+
+    public function openNotificationMenu(string $expectedDescription): void
+    {
+        $this->getHTMLPage()
```

Comment:
This is not necessary imo, you are checking if table item exists while clicking on the notification item menu.